Topics/Development_Process/App_Store_Deployment.md

```diff
@@ -0,0 +1,144 @@
+
+3. In the Certificate Assistant dialog, enter an email address in the User Email Address field.
+
+4. In the Common Name field, enter a name for the key (for example, Gita Kumar Dev Key).
+
+5. Leave the CA Email Address field empty.
+
+6. Choose “Saved to disk,” then click Continue.
+
+Now you can upload this CSR onto the website to generate your certificate.
+
+The full details on all of Apple's Certifications are available on the Apple Certificates page: 
+[Apple Certificates](https://developer.apple.com/support/certificates/)
+
+
+### Beta Testing
+Prior to releasing the app for review, it would be good practice to beta test the app
+for any possibly new bugs.
+
+**For exampel:** a common error that many people find at this stage is with the push notifications. 
+As this requires a certificate of its own, the first step should be to verify that the certificate is valid and that you have handled the distribution part of this testing stage correctly.
+
+In terms of beta testing, the two main choices are:
+
+#### TestFlight
+TestFlight can be considered the 'App Store' for beta testing your apps. 
+It is considered its own 'app market' and as the developer(s), you can decide whether to release your app for public testing, or keep it internal for select testers. This makes use of App Store connect, which eases the process of uploading your apps onto the App Store. 
+From there, you will be able to submit your app to both TestFlight and the App Store if you'd like. 
+You should also make note of the fact that submitting an app onto TestFlight will also go through a review process just like the App Store, albeit with more lenient standards which could result in a quicker review time.
+
+#### Ad-Hoc
+Ad-Hoc allows you to distribute your app to a limited number (100) of registered devices. 
+Unlike TestFlight, this cannot release it for 'public testing' to all beta testers, so you would have to register the devices you wish to use in the devices section on the Apple Developer Page. This means that you would have to register their device first by entering it's UDID (Unique Device Identifier) onto the App Developer page. 
+
+- This can be found by plugging in your iPhone to your MacBook, going onto Xcode, then clicking Window tab to Devices and Simulators. 
+From there you can select your iPhone and copy its UDID (make sure to include the '-' that is found after the 8th character).
+
+The benefit of using Ad hoc is that you would not need the app to be reviewed before distributing it out for testing. This means that you would be able to get your app to your testers immediately and possibly avoid getting the app rejected during the review process.
+
+#### Distribution Process
+After your generate all your needed certificates, you would then need to generate your Provisioning Profile, selecting the app you wish to use as well as all the devices you want to access it. 
```

Comment:
What is a 'Provisioning Profile'? From context it seems like it is where you choose how to distribute your application, but it could be clearer.